0.data_analysis_and_processing/mic_analysis/nbconverted/treated_well_sirna_mic_analysis.py

```diff
@@ -0,0 +1,248 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Evaluate Well Stain Relationship Strength
+# This analysis compares Maximal Information Coefficient (MIC) scores between the same DAPI and GOLD aggregated well features per well.
+# Distributions of these MIC scores are visualized, between zero and one, where one indicates a perfect relationship and zero indicates no relationship.
+
+# In[1]:
+
+
+import pathlib
+import re
+import sys
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+
+
+# ## Find the root of the git repo on the host system
+
+# In[2]:
+
+
+# Get the current working directory
+cwd = pathlib.Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# # Custom Imports
+
+# In[3]:
+
+
+sys.path.append(f"{root_dir}/0.data_analysis_and_processing/utils")
+
+from MIC import MIC
+from PairwiseCompare import PairwiseCompare
+from ShuffledMIC import ShuffledMIC
+
+
+# # Inputs
+
+# In[4]:
+
+
+# Paths to original nuclear speckle data
+data_dir = root_dir / "nuclear_speckles_data"
```

Comment:
Disregard if this is different than the repository since the mask data is only accessible after performing the CP pipeline, not uploaded to GitHub. 

We probably will decide to upload the data at another time where it will be easier to reproduce.